actions: enable code scanning

FrostyX opened the pull request:
The lgtm.com is now dead, let's migrate to GitHub's built-in code-scanning feature

https://github.blog/2020-09-30-code-scanning-is-now-available/ 
https://docs.github.com/en/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/setting-up-code-scanning-for-a-repository

FrostyX commented:
The `.github/workflows/codeql.yml` file was generated by project settings, I only removed java from it.

It found a couple of errors. Some of them I am fixing in this PR, and some of them I waived as false-positives
https://github.com/fedora-copr/copr/security/code-scanning?query=pr%3A2464+tool%3ACodeQL+is%3Aclosed
I am not quite sure if the waiving is per PR or if it will be remembered. We will see



I think we can use this badge for our README: [![CodeQL](https://github.com/fedora-copr/copr/actions/workflows/codeql.yml/badge.svg)](https://github.com/fedora-copr/copr/actions/workflows/codeql.yml)
I want to get the GitHub action merged first.

praiskup commented:
Not necessarily a blocker, but can you please check if this is the way to go WRT ESSv9?

praiskup commented on keygen/tests/test_logic.py:
```diff
@@ -181,3 +183,14 @@ def test_tmpfiles_errors(self, popen, user_exists):
+
+    def test_validate_name_email(self, _popen, _user_exists):
+        assert logic.validate_name_email("frostyx#foo@copr.fedorahosted.org")
+        assert logic.validate_name_email("@copr#foo@copr.fedorahosted.org")
```
Just note: https://bugzilla.redhat.com/show_bug.cgi?id=1813244 and #1320

Perhaps we should add a small TODO at least?

praiskup commented:
Can we use some markdown badges similar to what we had before #2462?

FrostyX commented:
I updated the keygen validator code and added some asserts to tests.

Just for the record, it found one `Medium` warning, which didn't mark the CI as failed. It seems that only `High` and `Critical` does that.

praiskup commented on keygen/src/copr_keygen/logic.py:
```diff
@@ -37,6 +38,34 @@
+    # validating email is too hardcore (see `wtforms.validators.Email`) and we
+    # don't care anyway. We only want to make sure that no ilegal characters
+    # were used.
+    if not re.match(r"^[\w.-@]+$", email):
```
What a nice catch.
